src/FixerConfiguration/FixerOption.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\FixerConfiguration;
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+
+final class FixerOption implements FixerOptionInterface
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $description;
+
+    /**
+     * @var mixed
+     */
+    private $default;
+
+    /**
+     * @var bool
+     */
+    private $useDefault = false;
+
+    /**
+     * @var array|null
+     */
+    private $allowedTypes;
+
+    /**
+     * @var array|null
+     */
+    private $allowedValues;
+
+    /**
+     * @var \Closure|null
+     */
+    private $normalizer;
+
+    /**
+     * @param string $name
+     * @param string $description
+     */
+    public function __construct($name, $description)
+    {
+        $this->name = $name;
+        $this->description = $description;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return $this->description;
+    }
+
+    /**
+     * @param mixed $default
+     *
+     * @return $this
+     */
+    public function setDefault($default)
+    {
+        $this->default = $default;
+        $this->useDefault = true;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasDefault()
+    {
+        return $this->useDefault;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefault()
+    {
+        if (!$this->hasDefault()) {
+            throw new \LogicException('No default value defined.');
+        }
+
+        return $this->default;
+    }
+
+    /**
+     * @param string[]|string|null $allowedTypes
```

Comment:
no need for mimics symfony here. i prefer to be strict, then Zend releases PHP 7 which allows for that as well ;)
symfony code was created ages ago